File: bgm-code/ParameterEstimation_Stat/ExamsQuizzes/ExamsQuizzes_1.R
# clears workspace:  
rm(list=ls(all=TRUE)) 

# sets working directories:
#setwd("C:/Documents and Settings/ewagenmakers/Desktop/temp/BayesBook")
setwd("C:/Documents and Settings/ewagenmakers/My Documents/My Dropbox/EJ/temp/BayesBook/Code/ParameterEstimation_Stat/ExamsQuizzes")
library(R2WinBUGS)
bugsdir = "C:/Program Files/WinBUGS14"

k = c(21,17,21,18,22,31,31,34,34,35,35,36,39,36,35)
p = length(k) #number of people
n = 40 # number of questions


data  = list("p", "k", "n") # to be passed on to WinBUGS
myinits =	list(
  list(phi = c(NA,0.5), z = round(runif(p)))) # Initial group assignment

# parameters to be monitored:	
parameters = c("phi","z")

# The following command calls WinBUGS with specific options.
# For a detailed description see Sturtz, Ligges, & Gelman (2005).
samples = bugs(data, inits=myinits, parameters,
	 			model.file ="ExamsQuizzes_1.txt",
	 			n.chains=1, n.iter=1000, n.burnin=0, n.thin=1,
	 			DIC=T, bugs.directory=bugsdir,
	 			codaPkg=F, debug=T)
# Now the values for the monitored parameters are in the "samples" object, 
# ready for inspection.

File: bgm-code/ParameterEstimation_Stat/ExamsQuizzes/ExamsQuizzes_1.m
% Exam Scores

clear;

% Data
k=[21 17 21 18 22 31 31 34 34 35 35 36 39 36 35];n=40;
% Number of people
p=length(k);

% Sampling Parameters
nchains=1; % number of chains
nburnin=0; % number of burn-in samples
nsamples=1e3; % number of samples

% Data to Supply to WinBugs
datastruct = struct('p',p,'k',k,'n',n);

% Initial Values to Supply to WinBugs
for i=1:nchains
    S.z = round(rand(1,p)); % Initial Random Group Assignments
    init0(i) = S;
end

[samples, stats] = matbugs(datastruct, ...
    fullfile(pwd, 'ExamsQuizzes_1.txt'), ...
    'init', init0, ...
    'nChains', nchains, ...
    'view', 1, 'nburnin', nburnin, 'nsamples', nsamples, ...
    'thin', 1, 'DICstatus', 0, 'refreshrate',100, ...
    'monitorParams', {'phi','z'}, ...
    'Bugdir', 'C:/Program Files/WinBUGS14');



